File: app/src/main/java/com/example/home/db/HitsItemDao.java
package com.example.home.db;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.home.model.HitsItem;

import java.util.ArrayList;
import java.util.List;

@Dao
public interface HitsItemDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void addHitsItem(HitsItem hitsItem);

    @Query("select * from HitsItem")
    List<HitsItem> getData();
}

File: app/src/main/java/com/example/assignment13_mvvm_kotlin/db/HitsItemDao.kt
package com.example.assignment13_mvvm_kotlin.db

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.assignment13_mvvm_kotlin.model.HitsItem

@Dao
interface HitsItemDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertHitsItem(hitsItem: HitsItem)

    @Query("select * from HitsItem")
    suspend fun getSavedHits() : List<HitsItem>

}